src/NServiceBus.Core/ConsistencyGuarantees/ConsistencyGuarantee.cs

```diff
@@ -0,0 +1,9 @@
+namespace NServiceBus.ConsistencyGuarantees
+{
+    /// <summary>
+    /// Base class for the various supported consistency guarantees
+    /// </summary>
+    public abstract class ConsistencyGuarantee
```

Comment:
Not sure, we will some kind of ordering btw them and I'm not sure it's linear. @SzymonPobiega @danielmarbach thoughts?

I'm good with YAGNI on that and use an enum for now though